### PR TITLE
:bug:Source Twitter: Improve Query Parameter Specification

### LINKED ISSUE #25415: Source Twitter: Improve query parameter specification
## Tell us about the problem you're trying to solve
The current Source Twitter Connector retrieves only some specific and hard-coded fields of the sent requests. Users should be able to specify all the parameters they require for the query instead of being locked into hard-coded ones.

## Describe the solution you’d like
Being able to specify all the other possible parameters for the query as described in the [docs](https://developer.twitter.com/en/docs/twitter-api/fields): `tweet.fields`, `user.fields`,  `media.fields`, ... 

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by topefolorunso**
## What
Fixes #25415

## How
*Describe the solution*

## Recommended reading order
1. `manifest.yaml`
2. `spec.yaml`
3. `tweets.json`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml [hunk 1]
```diff
@@ -13,7 +13,11 @@ definitions:
       api_token: "{{ config.api_key }}"
     request_parameters:
       query: "{{ config['query'] }}"
-      tweet.fields: "author_id,conversation_id,created_at,in_reply_to_user_id,lang"
+      tweet.fields: '{{ ",".join(config["tweet_fields"]) }}'
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml:16]: if the user has old config what next? 


look like braking changes

**Review by lazebnyi — CHANGES_REQUESTED** [review]
need bump version,  docs and change log 

btw need provide backward compatibility

File: airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml (hunk 1 above)
Comment (topefolorunso) [reply airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml:16]: Yes, it is a breaking change. User now has to specify the fields that they want in the config - https://github.com/airbytehq/airbyte/issues/25415#issue-1680084909

**Comment by topefolorunso** [comment]
> need bump version, docs and change log

@lazebnyi Done

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml (hunk 1 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml:16]: yes, that's can guarantee backward compatibility

Comment (lazebnyi) [reply airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml:16]: yes, that's can guarantee backward compatibility

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml [hunk 2]
```diff
@@ -13,7 +13,11 @@ definitions:
       api_token: "{{ config.api_key }}"
     request_parameters:
       query: "{{ config['query'] }}"
-      tweet.fields: "author_id,conversation_id,created_at,in_reply_to_user_id,lang"
+      tweet.fields: '{{ ",".join(config["tweet_fields"], 'author_id,conversation_id,created_at,in_reply_to_user_id,lang') }}'
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-twitter/source_twitter/manifest.yaml:16]: Will that break the request if user add any field from the default list?

File: docs/integrations/sources/twitter.md [hunk 3]
```diff
@@ -16,7 +16,12 @@ To set up the Twitter source connector, you'll need the [App only Bearer Token](
 6. For **Search Query**, enter the query for matching Tweets. You can learn how to build this query by reading [build a query guide](https://developer.twitter.com/en/docs/twitter-api/tweets/search/integrate/build-a-query).
 7. For **Start Date (Optional)**, enter the date in `YYYY-MM-DDTHH:mm:ssZ` format. The start date for retrieving tweets cannot be more than 7 days in the past.
 8. For **End Date (Optional)**, enter the date in `YYYY-MM-DDTHH:mm:ssZ` format. The end date for retrieving tweets must be a minimum of 10 seconds prior to the request time.
-9. Click **Set up source**.
+9.  For **tweet_fields**, select the tweet fields you want to extract from the API. See [here](https://developer.twitter.com/en/docs/twitter-api/fields) for more details.
```
Comment (lazebnyi) [inline docs/integrations/sources/twitter.md:19]: Please add info that we have a default list for tweet fields.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
@topefolorunso closing this due inactivity

**Pull request closed without merging**